cliquet/initialization.py

```diff
@@ -373,15 +374,45 @@ def on_new_response(event):
     config.add_subscriber(on_new_response, NewResponse)
 
 
+def _filter_events(callback, actions, resources):
+    def wrapped(event):
+        action = event.payload.get('action')
+        resource_name = event.payload.get('resource_name')
+        if action and action not in actions:
+            return
+        if resource_name and resources and resource_name not in resources:
+            return
+        return callback(event)
+    return wrapped
+
+
 def setup_listeners(config):
-    listeners = aslist(config.get_settings()['event_listeners'])
-    from cliquet.events import ResourceChanged
+    write_actions = (ACTIONS.CREATE, ACTIONS.UPDATE, ACTIONS.DELETE)
+    settings = config.get_settings()
+    listeners = aslist(settings['event_listeners'])
 
-    for listener in listeners:
+    for name in listeners:
         logger.info('Setting up %r listener')
-        listener_mod = config.maybe_dotted(listener)
-        listener = listener_mod.load_from_config(config)
-        config.add_subscriber(listener, ResourceChanged)
+        prefix = 'event_listeners.%s.' % name
+
+        try:
+            listener_mod = config.maybe_dotted(name)
+            assert hasattr(listener_mod, 'load_from_config')
```

Comment:
why a plain assert here ? it means that if we run python using the -O option this line is stripped
I think you just want to catch AttributeError here